openmc/lattice.py

```diff
@@ -1094,18 +1180,48 @@ def get_local_coordinates(self, point, idx):
             system
 
         """
-        x = point[0] - (self.center[0] + sqrt(0.75)*self.pitch[0]*idx[0])
-        y = point[1] - (self.center[1] + (0.5*idx[0] + idx[1])*self.pitch[0])
+        if self._hextype:
+            x = point[0] - (self.center[0] + self.pitch[0]*idx[0] + 
+                            0.5*self.pitch[0]*idx[1])
+            y = point[1] - (self.center[1] + 
+                            (sqrt(0.75)*self.pitch[0]*idx[1]))
+        else:
+            x = point[0] - (self.center[0] 
+                            + sqrt(0.75)*self.pitch[0]*idx[0])
+            y = point[1] - (self.center[1] 
+                            + (0.5*idx[0] + idx[1])*self.pitch[0])
+
         if self._num_axial is None:
             z = point[2]
         else:
             z = point[2] - (self.center[2] + (idx[2] + 0.5 - 0.5*self.num_axial)*
                             self.pitch[1])
         return (x, y, z)
-
+                       
```

Comment:
Remove trailing whitespace (I'd recommend configuring your text editor to do this by default)